accord-core/src/main/java/accord/utils/async/AsyncChains.java

```diff
@@ -253,8 +415,37 @@ public static abstract class FlatMap<I, O> extends Link<I, O> implements Functio
         @Override
         public void accept(I i, Throwable throwable)
         {
-            if (throwable != null) next.accept(null, throwable);
-            else apply(i).begin(next);
+            try
+            {
+                if (throwable != null) next.accept(null, throwable);
+                else                   apply(i).begin(next);
+            }
+            catch (Throwable t)
+            {
+                throw t;
+            }
+        }
+    }
+
+    private static abstract class DebugFlatMap<I, O> extends Link.Debug<I, O> implements Function<I, AsyncChain<O>>
```

Comment:
similarly: wouldn't it be better to extend `FlatMap` since it has more bespoke logic we may modify in future and want to inherit any changes?

Reply:
it might even be better to just have a single DebugWrapper object, that wraps all of the results of each call? We pay an additional object overhead per link but only when debug is enabled, and don't have to extend anything?